app/src/main/java/com/example/polina/socialnetwork/SearchActivity.java

```diff
@@ -12,67 +14,79 @@
+
+public class SearchActivity extends Activity {
 
     SNApp snApp;
-    Context thisContext;
     ArrayList<User> users = new ArrayList<>();
     SearchAdapter adapter;
     EditText searchName;
+    private SharedPreferences sharedPreferences;
 
 
-
-    public View onCreateView(LayoutInflater inflater, ViewGroup container,
-                             Bundle savedInstanceState) {
-        View v = inflater.inflate(R.layout.fragment_search, null);
-        snApp= (SNApp) getActivity().getApplication();
-        searchName = (EditText) v.findViewById(R.id.search_name);
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        sharedPreferences = getSharedPreferences(Utils.PROFILE_PREFERENCES, MODE_PRIVATE);
+        setContentView(R.layout.search_activity);
+        snApp = (SNApp) getApplication();
+        searchName = (EditText) findViewById(R.id.search_name);
         searchName.setOnEditorActionListener(new TextView.OnEditorActionListener() {
             @Override
             public boolean onEditorAction(TextView textView, int i, KeyEvent keyEvent) {
                 System.err.println(" press onclick");
-                if(i== EditorInfo.IME_ACTION_SEARCH) {
+                if (i == EditorInfo.IME_ACTION_SEARCH) {
                     new LoadUsers().execute(searchName.getText().toString());
                     return true;
                 }
                 return false;
             }
         });
-        thisContext = container.getContext();
-        ListView searchList = (ListView)v.findViewById(R.id.search_result);
-        adapter = new SearchAdapter(users, thisContext, snApp.mImageLoader);
+
+        ListView searchList = (ListView) findViewById(R.id.search_result);
+        adapter = new SearchAdapter(users, this, snApp.mImageLoader);
         searchList.setAdapter(adapter);
         searchList.setOnItemClickListener(new AdapterView.OnItemClickListener() {
             @Override
             public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
                 User user = (User) adapterView.getItemAtPosition(i);
                 System.err.println(" user ID" + user.userId);
+                Intent intent = new Intent(SearchActivity.this, UserActivity.class);
+                if(user.userId.equals(sharedPreferences.getString(Utils.ID, ""))){
+                    intent.putExtra(Utils.USER_ID, Utils.MY_PROFILE);
+                } else {
+                    intent.putExtra(Utils.USER_ID, user.userId);
+                }
+                startActivity(intent);
             }
         });
 
-        return v;
+
     }
 
 
-    class LoadUsers extends AsyncTask<String, Void, JSONObject>{
+    class LoadUsers extends AsyncTask<String, Void, JSONObject> {
 
         @Override
         protected JSONObject doInBackground(String... strings) {
-            if(!strings[0].isEmpty()) {
+            if (!strings[0].isEmpty()) {
```

Comment:
never use String.isEmpty(). it doesn't guaranty the we won't get NPE. Use static method TextUtils.isEmpty(str)